dali/operators/image/resize/resize_base.cc

```diff
@@ -16,164 +16,117 @@
+      (this->template SetupResizeTyped<float, InputType>(out_shape, in_shape, params,
+        spatial_ndim, first_spatial_dim)),
+      (DALI_FAIL(make_string("Unsupported type: ", in_type,
+        ". Supported types are: uint8, int16, uint16 and float"))));
   }
-#undef DALI_MAP_INTERP_TO_RESAMPLE
 }
 
-DALI_SCHEMA(ResamplingFilterAttr)
-  .DocStr(R"code(Resampling filter attribute placeholder)code")
-  .AddOptionalArg("interp_type",
-      R"code(Type of interpolation used. Use `min_filter` and `mag_filter` to specify
-different filtering for downscaling and upscaling.)code",
-      DALI_INTERP_LINEAR)
-  .AddOptionalArg("mag_filter", "Filter used when scaling up",
-      DALI_INTERP_LINEAR)
-  .AddOptionalArg("min_filter", "Filter used when scaling down",
-      DALI_INTERP_LINEAR)
-  .AddOptionalArg("temp_buffer_hint",
-      "Initial size, in bytes, of a temporary buffer for resampling.\n"
-      "Ingored for CPU variant.\n",
-      0)
-  .AddOptionalArg("minibatch_size", "Maximum number of images processed in a single kernel call",
-      32);
-
-ResamplingFilterAttr::ResamplingFilterAttr(const OpSpec &spec) {
-  DALIInterpType interp_min = DALIInterpType::DALI_INTERP_LINEAR;
-  DALIInterpType interp_mag = DALIInterpType::DALI_INTERP_LINEAR;
-
-  if (spec.HasArgument("min_filter"))
-    interp_min = spec.GetArgument<DALIInterpType>("min_filter");
-  else if (spec.HasArgument("interp_type"))
-    interp_min = spec.GetArgument<DALIInterpType>("interp_type");
-
-  if (spec.HasArgument("mag_filter"))
-    interp_mag = spec.GetArgument<DALIInterpType>("mag_filter");
-  else if (spec.HasArgument("interp_type"))
-    interp_mag = spec.GetArgument<DALIInterpType>("interp_type");
-
-  min_filter_ = { interp2resample(interp_min), 0 };
-  mag_filter_ = { interp2resample(interp_mag), 0 };
-
-  temp_buffer_hint_ = spec.GetArgument<int64_t>("temp_buffer_hint");
-}
-void ResizeBase::SubdivideInput(const kernels::InListGPU<uint8_t, 3> &in) {
-  for (auto &mb : minibatches_) {
-    sample_range(mb.input, in, mb.start, mb.start + mb.count);
-  }
+template <typename Backend>
+template <typename OutputType, typename InputType>
+void ResizeBase<Backend>::SetupResizeTyped(
+      TensorListShape<> &out_shape,
+      const TensorListShape<> &in_shape,
+      span<const kernels::ResamplingParams> params,
+      int spatial_ndim,
+      int first_spatial_dim) {
+  VALUE_SWITCH(spatial_ndim, static_spatial_ndim, (2),
+  (SetupResizeStatic<OutputType, OutputType, static_spatial_ndim>(
+      out_shape, in_shape, params, first_spatial_dim)),
+  (DALI_FAIL(make_string("Unsupported number of resized dimensions: ", spatial_ndim))));
 }
 
-void ResizeBase::SubdivideOutput(const kernels::OutListGPU<uint8_t, 3> &out) {
-  for (auto &mb : minibatches_) {
-    sample_range(mb.output, out, mb.start, mb.start + mb.count);
+template <>
+template <typename OutputType, typename InputType, int spatial_ndim>
+void ResizeBase<GPUBackend>::SetupResizeStatic(
```

Comment:
The only difference compare to void `ResizeBase<CPUBackend>::SetupResizeStatic` I see is  is just the `ImplType`. Can you parameterize it as well?

Reply:
The difference is in the parameters passed to make_unique - minibatch_size for GPU and num_threads for CPU.